docs/001_develop/03_client-capabilities/023_custom-components/05_directives

```diff
@@ -0,0 +1,291 @@
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+@customElement({
+    name: "my-button",
+    template: html<MyButton>`
+        <div>
+            ${when(
+                x => !x.ready,
+                html<MyButton>`Loading...`
+            )}
+            ${when(
+                x => x.ready,
+                html<MyButton>`<button>Data Loaded: ${x => x.data.message}</button>`
+            )}
+        </div>
+    `,
+})
+export class MyButton extends FASTElement {
+     @observable ready: boolean = false;  # Track loading state
+     @observable data: { message: string } = null; # Store hardcoded data
+
+    connectedCallback() {
+        super.connectedCallback();
+        console.log("connectedCallback called");
+        this.simulateDataLoad();
+    }
+
+    async simulateDataLoad() {
+        console.log("simulateDataLoad called");
+        await delay(2000); # 2-second delay
+        this.data = { message: "Hello, Genesis!" }; # Hardcoded data
+        this.ready = true; # Indicate loading is complete
+    }
+}
+```
+
+Initial State:
+
+- When `<my-button>` is first added to the DOM, it displays "Loading..." for 2 seconds (simulated delay).
+
+After 2 Seconds:
+
+- Once the delay completes, ready is set to true, and the button appears with the text "Data Loaded: Hello, Genesis!".
+
+### Syntax breakdown
+
+```shell
+when(condition, template);
+```
+
+- Condition: A boolean expression that determines if the template should be rendered. If true, the template is added to the DOM; if false, it’s removed.
+- Template: Defines the HTML structure to render when the condition is true.
+
+## whenElse
+
+In the previous example with `when`, we demonstrated how to handle conditional rendering for 'if-else' scenarios by using `when` twice. However, there's a more efficient approach: the `whenElse` directive. With `whenElse`, you can manage if-else conditions in a single directive, avoiding the need to use `when` multiple times.
+
+ <!-- Look into how to implement this -->
+
+### Usage in HTML
+
+## Repeat
+
+The `repeat` directive is used to dynamically render a list of items. This directive is highly useful for rendering lists, especially when you need to manage and display data dynamically.
+
+`<my-button-list>` will display a static list of button labels, each rendered as a button. This example shows how to use repeat to render each item in the list.
+
+```shell
+import { GenesisElement, customElement, observable, html, repeat } from "@genesislcap/web-core";
+
+@customElement({
+    name: "my-button-list",
+    template: html<MyButtonList>`
+        <h2>Button List</h2>
+        <ul>
+            ${repeat(x => x.buttons, html<string>`
+                <li><button>${x => x}</button></li>
+            `)}
+        </ul>
+    `,
+})
+export class MyButtonList extends GenesisElement {
+    @observable buttons: string[] = ["Click Me", "Submit", "Reset"]; // Static list of button labels
```

Comment:
```suggestion
    buttons: string[] = ["Click Me", "Submit", "Reset"]; // Static list of button labels
```
This doesn't need to be observable for your example, and the way observable works with array is [a bit tricky](https://fast.design/docs/1.x/fast-element/observables-and-state#observing-arrays) so I wouldn't just use the keyword here without an explanation